pkg/cli/gettoken/tokencache/tokencache.go

```diff
@@ -0,0 +1,90 @@
+package tokencache
+
+import (
+	"crypto/sha256"
+	"encoding/gob"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+type Set struct {
+	IDToken      string
+	RefreshToken string
+}
+
+type Key struct {
+	IssuerURL      string
+	ClientID       string
+	ClientSecret   string
+	Username       string
+	ExtraScopes    []string
+	CACertFilename string
+	CACertData     string
+	SkipTLSVerify  bool
```

Comment:
I'm not sure about the exact reasons, it is copied from here https://github.com/int128/kubelogin/blob/79a6f8596774d9d1fff2319ddc84a3a87f80fa48/pkg/tokencache/types.go and I removed some very unrelated ones such as `Username`. 

But it makes sense having such granularity because user may take different token with different CA cert or with skipping tls verify. There are 2 tokens and user can use the one when tls verification is skipped, and the other one when passing different CA cert.

Reply:
if a user grabs a different token with a different CA, it's still a token for the user from the same issuer with the same client